src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const joinedArr = [];
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (i === this.length - 1) {
+        joinedArr[i] = this[i];
+        break;
+      }
+
+      if (separator === null) {
+        joinedArr[i] = this[i] + 'null';
+      } else {
+        joinedArr[i] = this[i] + separator;
+      }
```

Comment:
you don't need it, `null` value is appended itself, gets converted into `"null"`